tenets.md

```diff
@@ -0,0 +1,423 @@
+than it is written
+so we prioritize the expression of intent
+for those who come after us.
+By structuring clarity at the call site
+we maintain the readability of our codebase.
+By structuring our API
+to encourage this same kind of clarity
+we help customers express their intent
+when using our tools.
+
+Examples of features used to express clear inputs
+at the call site are
+named parameters, enums, and option hash.
+
+## Motivation
+
+Though expressing intent at the call site
+makes the AWS Encryption SDK (ESDK) more verbose
+it makes function calls obviously correct by construction.
+
+As a developer gets more experience
+with a library they will remember
+what parameters do what and how they are passed.
+But this requires time and effort.
+
+Due to the critical nature
+of the data handled by the AWS Encryption SDK
+this extensive experience MUST NOT
+be a requirement for review.
+The reviewer, auditor,
+or a later developer performing maintenance
+SHOULD be able to easily understand
+the intent of the parameters that are passed.
+
+This is important because
+while the complexity of systems generally increase,
+there is no requirement that
+the experience and understanding
+of the people maintaining these system must increase.
+Just because something is crystal clear to you today
+does not mean it will be equally clear to you later.
+Let alone someone without your experience and understanding.
+
+We express this in the specification
+by naming options and encouraging enums.
+But also our examples MUST use
+expressive variable names
+to help drive home the intention.
+
+## Sensible Defaults
+
+## Summary
+
+Functions need parameters to know what actions to take.
+Sometimes these parameters should have defaults.
+A parameter that has a default
+will take a value specific if no value is passed.
+
+This simplifies using the function
+because callers do not need
+to select a value.
+
+An example is the principle of least surprise.
+Customers MUST have the freedom consider,
+but SHOULD NOT need extensive domain understanding
+to get started.
+
+Defaults should guide customers towards
+the best in class configurations that we know of.
+As customers need to consider tradeoffs,
+they can do so in an informed manner.
+In this way another example might be
+nudge theory from behavioral economics.
+
+## Motivation
+
+The AWS Encryption SDK (ESDK) aims
+to be [easy to use](#easy-to-use).
+Limiting the number of options
+that are needed lowers the barrier to entry.
+Sensible Defaults should be selected
+where possible.
+
+Sensible Defaults are defaults
+that are optimal in the majority of use cases.
+If a default is optimal in all use cases,
+then it dominates all other options.
+These kinds of dominate options
+SHOULD NOT be configurable at all.
+
+Given the relationship between encrypt and decrypt
+options are preferred on encrypt.
+Once a message has been encrypted,
+decryption of that message
+SHOULD BE gated by access control and correctness,
```

Comment:
np: here and below, BE doesn't need to be capitalized

Reply:
```suggestion
SHOULD be gated by access control and correctness,
```